### PR TITLE
add type annotations to conv_fused/blas_compare/blas_compare_setup

### LINKED ISSUE #51234: Enable conv_fused/blas_compare/blas_compare_setup typecheck during CI
Enable typecheck for the following files:
- torch._overrides
- torch.utils.tensorboard._caffe2_graph
- torch.contrib._tensorboard_vis
- torch.nn.utils.prune
- torch.utils.show_pickle
- torch.utils.hipify.hipify_python
- torch.utils.benchmark.examples.*
- torch.nn.quantized.modules.batchnorm
- torch.nn.intrinsic.qat.modules.conv_fused

cc @ezyang @malfet @rgommers @xuzhao9 @gramster

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Fixes #51234

### CODE REVIEW

File: torch/nn/intrinsic/qat/modules/conv_fused.py [hunk 1]
```diff
@@ -86,6 +91,7 @@ def freeze_bn_stats(self):
         return self
 
     def _forward(self, input):
+        assert isinstance(self.bn.running_var, torch.Tensor)
```
Comment (rgommers) [inline torch/nn/intrinsic/qat/modules/conv_fused.py:94]: I remembered there was an issue with this in `BatchNorm`, but here this is correct because `self.track_running_stats` is always True. So this is fine.

File: torch/nn/intrinsic/qat/modules/conv_fused.py [hunk 2]
```diff
@@ -99,6 +105,9 @@ def _forward(self, input):
             zero_bias = torch.zeros_like(self.bias)
         else:
             zero_bias = torch.zeros(self.out_channels, device=scaled_weight.device)
+        # mypy error: "Tensor" not callable  [operator]
+        # self._conv_forward is a method
+        assert not isinstance(self._conv_forward, torch.Tensor)
```
Comment (rgommers) [inline torch/nn/intrinsic/qat/modules/conv_fused.py:110]: Weird this is needed, Mypy cannot make sense of the inherited `nni._FusedModule` it looks like. Also fine.

**Review by rgommers — APPROVED** [review]
LGTM, thanks @guilhermeleobas. The one CI failure is unrelated (it's also showing up in other PRs), this should be good to land.

File: torch/nn/intrinsic/qat/modules/conv_fused.py (hunk 2 above)
Comment (walterddr) [reply torch/nn/intrinsic/qat/modules/conv_fused.py:110]: Hi @guilhermeleobas, I remember there were some mypy caching issues previously. did you try deleting the mypy cache and rerun test/test_type_hints.py without this assert?

**Comment by walterddr** [comment]
based on a quick research #51473. --> it looks like conv.py is missing a function signature. i will amend the commit and merge it once CI shows green.

**Review by malfet — APPROVED** [review]

**Comment by guilhermeleobas** [comment]
Thanks for the change @walterddr
